Rubberduck.Refactorings/Common/DeclarationExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Refactorings.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.Common
+{
+    public static class DeclarationExtensions
+    {
+        public static bool IsVariable(this Declaration declaration)
+            => declaration.DeclarationType.HasFlag(DeclarationType.Variable);
+
+        public static bool IsMemberVariable(this Declaration declaration)
+            => declaration.IsVariable() && !declaration.ParentDeclaration.IsMember();
+
+        public static bool IsLocalVariable(this Declaration declaration)
+            => declaration.IsVariable() && declaration.ParentDeclaration.IsMember();
+
+        public static bool IsLocalConstant(this Declaration declaration)
+            => declaration.IsConstant() && declaration.ParentDeclaration.IsMember();
+
+        public static bool HasPrivateAccessibility(this Declaration declaration)
+            => declaration.Accessibility.Equals(Accessibility.Private);
+
+        public static bool IsMember(this Declaration declaration)
+            => declaration.DeclarationType.HasFlag(DeclarationType.Member);
+
+        public static bool IsConstant(this Declaration declaration)
+            => declaration.DeclarationType.HasFlag(DeclarationType.Constant);
+
+        public static bool IsUserDefinedTypeField(this Declaration declaration)
+            => declaration.IsMemberVariable() && IsUserDefinedType(declaration);
+
+        public static bool IsUserDefinedType(this Declaration declaration)
+            => (declaration.AsTypeDeclaration?.DeclarationType.Equals(DeclarationType.UserDefinedType) ?? false);
+
+        public static bool IsEnumField(this Declaration declaration)
+            => declaration.IsMemberVariable() && (declaration.AsTypeDeclaration?.DeclarationType.Equals(DeclarationType.Enumeration) ?? false);
+
+        public static bool IsDeclaredInList(this Declaration declaration)
+        {
+            return declaration.Context.TryGetAncestor<VBAParser.VariableListStmtContext>(out var varList)
+                            && varList.ChildCount > 1;
+        }
+
+        /// <summary>
+        /// Generates a Property Member code block specified by the letSetGet DeclarationType argument.
+        /// </summary>
+        /// <param name="variable"></param>
+        /// <param name="letSetGetType"></param>
+        /// <param name="propertyIdentifier"></param>
+        /// <param name="accessibility"></param>
+        /// <param name="content"></param>
+        /// <param name="parameterIdentifier"></param>
+        /// <returns></returns>
+        public static string FieldToPropertyBlock(this Declaration variable, DeclarationType letSetGetType, string propertyIdentifier, string accessibility = null, string content = null, string parameterIdentifier = null)
```

Comment:
I'm not sure this type of work really belongs here as an extension method to `Declaration`... would the target object not always be a `VariableDeclaration`?

Reply:
Yes - should have been `VariableDeclaration`.  Once the dust settles surrounding `EncapsulateField` - I'll put together a PR for this.

Reply:
I think this does not belong into an extension at all. Instead, I thing there should be some `ICodeBulder` to build the standard code for declarations. That would make the standard easily testable and would also allow to specify different versions of the code by injecting another implementation.
The same applies to the extensions for module body elements, even more, actually.

Reply:
@MDoerner I was certain using an extension was a debatable location for this functionality - and I was hoping for either confirmation or some guidance to move it elsewhere - thanks.  As I understand your comment, the `ICodeBuilder`  instance would be delivered by CW? 

 As per my earlier comment, I was planning to wait for the coming `EncapsulateField` PR before introducing anything further changes.